### PR TITLE
Big Refactor

### PR DESCRIPTION
**Pull request by micedre**
* Creation of several module for each provider and improvement on
  dependency cycle
* separation in 3 big functionality :
   * configuration fetching by implemnting RealmProvider interface (and
     depending on property)
   * services for interface with clients (http, jms, or other)
   * store providers for the actuel writer or reader (see javadoc for
     how to use it)
* a distribution module to manage different distribution (jar, war,
  docker...)
* addition of the spotless-maven plugin to apply/check google-format
  java codestyle and licenses to all source files

### CODE REVIEW

**Review by Donatien26 — APPROVED** [review]

File: sugoi-api-model/src/main/java/fr/insee/sugoi/model/UserStorage.java [hunk 1]
```diff
@@ -1,68 +1,86 @@
...
-    public void setStore(Store connection) {
-        this.store = connection;
-    }
-
+  private String name;
+  private String userSource;
+  private String organizationSource;
+  private Map<String, String> properties;
+  private String readerType;
+  private String writerType;
```
Comment (clement-dufaure) [inline sugoi-api-model/src/main/java/fr/insee/sugoi/model/UserStorage.java:24]: ok, using Class<T extends ReaderStore> for bean selection implies knowing ReaderStore interface here in module model.
To be discussed

File: sugoi-api-rest-services/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+### Récupération profil
+
+fr.insee.sugoi.ldap.profils.url=ldap-externe-dpii.recette.insee.fr
+fr.insee.sugoi.ldap.profils.port=389
+fr.insee.sugoi.ldap.profils.branche=cn=profil-contact-WebServicesLdap\,ou=WebServicesLdap_Objets\,ou=WebServicesLdap\,ou=Applications\,o=insee\,c=fr
+
+fr.insee.sugoi.ldap.default.ldap.size.pool=10
+fr.insee.sugoi.ldap.default.username=uid=appli_webservicesldap\,ou=WebServicesLdap_Objets\,ou=WebServicesLdap\,ou=Applications\,o=insee\,c=fr
+fr.insee.sugoi.ldap.default.password=webservices
+fr.insee.sugoi.ldap.default.pool=10
```
Comment (clement-dufaure) [inline sugoi-api-rest-services/src/main/resources/application.properties:15]: ⚠️ maybe not

**Review by clement-dufaure — COMMENTED** [review]
Some comments can be added to issues and reviewed later

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/mapper/AddressLdapMapper.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    address.put(
+        "Ligne3", searchResultEntry.getAttributeValue("inseeAdressePostaleCorrespondantLigne3"));
+    address.put(
+        "Ligne4", searchResultEntry.getAttributeValue("inseeAdressePostaleCorrespondantLigne4"));
+    address.put(
+        "Ligne5", searchResultEntry.getAttributeValue("inseeAdressePostaleCorrespondantLigne5"));
+    address.put(
+        "Ligne6", searchResultEntry.getAttributeValue("inseeAdressePostaleCorrespondantLigne6"));
+    address.put(
+        "Ligne7", searchResultEntry.getAttributeValue("inseeAdressePostaleCorrespondantLigne7"));
```
[the remark this answers is not in the document]
Comment (micedre) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/mapper/AddressLdapMapper.java:37]: yes those mappers need to be put elsewhere?

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/mapper/RealmLdapMapper.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+/** ProfilContextMapper */
+@Component
+public class RealmLdapMapper {
+
+  public Realm mapFromSearchEntry(SearchResultEntry searchResultEntry) {
+    Realm realm = new Realm();
+    UserStorage userStorage = new UserStorage();
+    realm.setName(searchResultEntry.getAttributeValue("cn").split("_")[1]);
+    String[] inseeProperties = searchResultEntry.getAttributeValues("inseepropriete");
```
[the remark this answers is not in the document]
Comment (micedre) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/mapper/RealmLdapMapper.java:30]: to be done later ?

File: sugoi-api-rest-services/src/main/resources/application.properties (hunk 2 above)
Comment (micedre) [reply sugoi-api-rest-services/src/main/resources/application.properties:15]: good catch :)

**Comment by micedre** [comment]
For the way to retrieve ReaderStore and Writer Store for a realm, I'm inclined to leave it as is, and refactor it if needed later, what do you think @clement-dufaure  ?

**Comment by clement-dufaure** [comment]
> 
> 
> For the way to retrieve ReaderStore and Writer Store for a realm, I'm inclined to leave it as is, and refactor it if needed later, what do you think @clement-dufaure ?

Ok, so approved this PR implies adding issue about this

**Comment by micedre** [comment]
> 
> 
> > For the way to retrieve ReaderStore and Writer Store for a realm, I'm inclined to leave it as is, and refactor it if needed later, what do you think @clement-dufaure ?
> 
> Ok, so approved this PR implies adding issue about this

Done, as well as moving ldap mappers to ldap-utils module

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/mapper/RealmLdapMapper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (clement-dufaure) [reply sugoi-api-core/src/main/java/fr/insee/sugoi/core/mapper/RealmLdapMapper.java:30]: ok, added to issues

**Review by clement-dufaure — APPROVED** [review]

**Pull request merged**